### PR TITLE
fix: [CDS-42730]: Filtering support added for linked templates

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Adds support to show linked templates in template selector for deployment template and for resolution of deployment template to store stepTemplateRefs in context

#### Screenshots

NA

https://user-images.githubusercontent.com/96409598/192143291-2a212fa5-a7ac-4fd7-afc5-d1edb3d06580.mov

Screenshot 2022-09-26 at 7 15 45 PM
Screenshot 2022-09-26 at 7 15 29 PM

Screenshot 2022-09-26 at 7 15 52 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11440)

File: src/modules/70-pipeline/hooks/useAddStepTemplate.ts [hunk 1]
```diff
@@ -85,7 +92,8 @@ export function useAddStepTemplate(props: AddStepTemplate): AddStepTemplateRetur
         templateType: 'Step',
         allChildTypes,
         gitDetails,
-        storeMetadata
+        storeMetadata,
+        ...(!isEmpty(linkedTemplateRefs) ? { linkedTemplate: { identifiers: linkedTemplateRefs } } : {})
```
Comment (yc147) [inline src/modules/70-pipeline/hooks/useAddStepTemplate.ts:96]: 1. You need to pass `checkboxLabel`
2. Same thing will be required in pipeline studio RightDrawer as well

Comment (samarth-sharma31) [reply src/modules/70-pipeline/hooks/useAddStepTemplate.ts]: Done

**Comment by samarth-sharma31** [comment]
retrigger jest

**Comment by samarth-sharma31** [comment]
trigger cypress

**Review by yc147 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**